packages/api/tests/lambdas/publish-reports/test-index.js

```diff
@@ -0,0 +1,253 @@
+    }
+  };
+});
+
+test.afterEach.always(async (t) => {
+  const {
+    granuleSnsTopicArnEnvVarBefore,
+    pdrSnsTopicArnEnvVarBefore,
+    QueueUrl,
+    TopicArn
+  } = t.context;
+
+  process.env.granule_sns_topic_arn = granuleSnsTopicArnEnvVarBefore;
+  process.env.pdr_sns_topic_arn = pdrSnsTopicArnEnvVarBefore;
+
+  await aws.sqs().deleteQueue({ QueueUrl }).promise()
+    .catch(noop);
+  await aws.sns().deleteTopic({ TopicArn }).promise()
+    .catch(noop);
+});
+
+test.after.always(() => revertPublishReports());
+
+test.serial('handler() publishes a PDR and a granule to SNS', async (t) => {
+  const {
+    granuleId, pdrName, QueueUrl, executionEvent
+  } = t.context;
+
+  await publishReports.handler(executionEvent);
+
+  await testMessagesReceived(t, QueueUrl, granuleId, pdrName);
+});
+
+test.serial('publishReportSnsMessages() publishes a PDR and a granule to SNS', async (t) => {
+  const {
+    granuleId, pdrName, QueueUrl, cumulusMessage
+  } = t.context;
+
+  await publishReports.publishReportSnsMessages(cumulusMessage);
+
+  await testMessagesReceived(t, QueueUrl, granuleId, pdrName);
+});
+
+test.serial('publishReportSnsMessages() publishes a granule to SNS even if publishing the PDR fails', async (t) => {
+  const {
+    granuleId, QueueUrl, cumulusMessage
+  } = t.context;
+
+  delete cumulusMessage.payload.pdr.name;
+
+  await publishReports.publishReportSnsMessages(cumulusMessage);
+
+  await testMessagesReceived(t, QueueUrl, granuleId, null);
+});
+
+test.serial('publishReportSnsMessages() publishes a PDR to SNS even if publishing the granule fails', async (t) => {
+  const {
+    pdrName, QueueUrl, cumulusMessage
+  } = t.context;
+
+  delete cumulusMessage.payload.granules[0].granuleId;
+
+  await publishReports.publishReportSnsMessages(cumulusMessage);
+
+  await testMessagesReceived(t, QueueUrl, null, pdrName);
+});
+
+test.serial('publishGranuleRecord() publishes a granule record to SNS', async (t) => {
+  const { granuleId, QueueUrl } = t.context;
+
+  await publishReports.publishGranuleRecord({ granuleId });
+
+  await testMessagesReceived(t, QueueUrl, granuleId, null);
+});
+
+test.serial('publishGranuleRecord() does not throw an exception if publishing the granule record to SNS fails', async (t) => {
+  const { granuleId } = t.context;
+
+  await t.notThrowsAsync(
+    () => publishReports.__with__({
+      publishSnsMessage: () => Promise.reject(new Error('nope'))
+    })(() => publishReports.publishGranuleRecord({ granuleId }))
+  );
+});
+
+test.serial('handleGranuleMessages() publishes a granule record to SNS', async (t) => {
+  const { cumulusMessage, granuleId, QueueUrl } = t.context;
+
+  await publishReports.handleGranuleMessages(cumulusMessage);
+
+  await testMessagesReceived(t, QueueUrl, granuleId, null);
+});
+
+test.serial('handlePdrMessage() publishes a PDR record to SNS', async (t) => {
+  const { cumulusMessage, pdrName, QueueUrl } = t.context;
+
+  delete cumulusMessage.payload.granules;
```

Comment:
This seems unnecessary?

Reply:
That's there to isolate this test so that it only creates a PDR.